dev/src/index.ts

```diff
@@ -1117,6 +1147,12 @@ export class Firestore {
         'All onSnapshot() listeners must be unsubscribed before terminating the client.'
       );
     }
+    if (this.bulkWritersCount > 0) {
+      return Promise.reject(
+        'All BulkWriter instances must be closed by calling and awaiting ' +
+          '`BulkWriter.close()` before terminating the client.'
+      );
+    }
```

Comment:
Optional suggestion: Could we combine the error message to something like:

"All onSnapshot() listeners must be unsubscribed and BulkWriter instances must be closed  before terminating the client. There are ${this.registeredListenersCount} active listeners and ${this. bulkWritersCount} that are still performing writes."

This gives more information and avoids the disappointment when the users have unregistered all listeners but still cannot terminate the client.

Reply:
combined.